base/statistics.jl

```diff
@@ -9,7 +9,7 @@ function mean(f::Callable, iterable)
     end
     count = 1
     value, state = next(iterable, state)
-    total = f(value)
```

Comment:
I couldn't think of a better way to ensure that the accumulator would be initialized to the appropriate type than to add `zero` with the return type of `f`. Unless you think it'd be better to make a temporary variable? Something like
```julia
f_value = f(value)
total = f_value + zero(f_value)
```